server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -400,19 +397,21 @@ private static boolean mergeTags(RulesDefinition.Rule ruleDef, RuleDto dto) {
     return removedRules;
   }
 
-  private void removeRule(DbSession session, List<RuleDto> removedRules, RuleDto rule) {
+  private void removeRule(DbSession session, List<RuleDefinitionDto> removedRules, RuleDefinitionDto rule) {
     LOG.info(String.format("Disable rule %s", rule.getKey()));
     rule.setStatus(RuleStatus.REMOVED);
-    rule.setSystemTags(Collections.<String>emptySet());
-    rule.setTags(Collections.<String>emptySet());
+    rule.setSystemTags(Collections.emptySet());
     update(session, rule);
+    // FIXME resetting the tags for all organizations must be handled a different way
+//    rule.setTags(Collections.emptySet());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)